objurl/objurl.go

```diff
@@ -243,7 +244,14 @@ func (o *ObjectURL) Match(key string) bool {
 }
 
 func (o *ObjectURL) String() string {
-	return o.Absolute()
+	if o.IsRemote() {
+		return o.Absolute()
+	}
+	return o.Base()
+}
+
+func (o *ObjectURL) MarshalJSON() ([]byte, error) {
+	return json.Marshal(o.String())
```

Comment:
`return []byte(o.String()), nil`

Reply:
It does not work same. It breaks to e2e tests :)

Reply:
Hmm, because of the quotes I guess.